File: HW/HW10/Python/LevelOrder.py
from collections import deque
from typing import Optional, List
from hw10 import TreeNode

class LevelOrder:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        lvl_order = []
        if root is None:
            return lvl_order
        queue = deque([root])
        while queue:
            size = len(queue)
            lvls = []
            for _ in range(size):
                node = queue.popleft()
                lvls.append(node.val)
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
            lvl_order.append(lvls)
        return lvl_order

File: HW/HW10/Java/LevelOrder.java
package HW10;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

public class LevelOrder {
	public List<List<Integer>> levelOrder(TreeNode root) {
		List<List<Integer>> lvlOrder = new ArrayList<>();
		if(root == null) {
			return lvlOrder;
		}
		Queue<TreeNode> queue = new LinkedList<>();
		queue.add(root);
		while(!queue.isEmpty()) {
			List<Integer> lvls = new ArrayList<>();
			int size = queue.size();
			for(int i = 0; i < size; i++) {
				TreeNode node = queue.poll();
				lvls.add(node.val);
				if(node.left != null) {
					queue.add(node.left);
				}
				if(node.right != null) {
					queue.add(node.right);
				}
			}
			lvlOrder.add(lvls);
		}
		return lvlOrder;
		
	}
}
